src/render/draw_symbol.js

```diff
@@ -236,7 +236,12 @@ function drawLayerSymbols(painter, sourceCache, layer, coords, isText, translate
     // Unpitched point labels need to have their rotation applied after projection
     const rotateInShader = rotateWithMap && !pitchWithMap && !alongLine;
 
-    const sortFeaturesByKey = layer.layout.get('symbol-sort-key').constantOr(1) !== undefined;
+    const canOverlap = layer.layout.get('icon-ignore-placement') ||
```

Comment:
This should probably be refactored into a separate function, but I'm not sure about the best place for it

Reply:
I think we should try add this to SymbolBucket's constructor to avoid the overhead of retrieving this value for every feature while it's guaranteed to be the same, and because there's already similar set up happening there (`this.sortFeaturesByKey`). In fact a very similar logic is already present there for `sortFeaturesByY`.

Reply:
Nice! I made `canOverlap` a field in `SymbolBucket`, and used it to calculate `sortFeaturesByY` as well.